chcemvediet/apps/inforequests/admin.py

```diff
@@ -315,3 +317,58 @@ def get_queryset(self, request):
         queryset = queryset.select_related(u'branch')
         queryset = queryset.select_related(u'email')
         return queryset
+
+    def get_actions(self, request):
+        actions = super(ActionAdmin, self).get_actions(request)
+        if u'delete_selected' in actions:
+            del actions[u'delete_selected']
+        return actions
+
+    def has_delete_permission(self, request, obj=None):
+        if obj is None:
+            return True
+        if obj.type in [Action.TYPES.REQUEST, Action.TYPES.ADVANCED_REQUEST]:
+            return False
+        if len(obj.branch.actions) > 1:
+            return True
+        return False
+
+    def render_delete_form(self, request, context):
+        using = router.db_for_write(self.model)
+        collector = admin.utils.NestedObjects(using)
+        collector.collect([context[u'object']])
+        to_delete = collector.nested()
+        for object in traverse(to_delete):
```

Comment:
@martinmacko47 Ako docielim, aby som upozornil nato, ze Akcia nie je posledna v historii. Template nie je vhodne prepisovat. Napadlo mi teda custom-formatovat prvy objekt z NestedObjects, teda akciu ktoru idem vymazat?

Reply:
@viliambalaz Idealne to vypis niekam do odrazky:

- Action: ...
   - Branch: ...
   - Branch: ...
- Outbound messages will be deleted:
    - Inforequest email: ...
    - Inforequest email: ...
- Inbound messages will be marked undecided:
    - Inforequest email: ...
    - Inforequest email: ...
- **Warning:** The deleted action is not the last action in the branch. Deleting it may cause logical errors in the inforequest history.

Reply:
@martinmacko47 Mam zadefinovat aj jazykove mutacie v `locales/django.po` alebo natvrdo napisat dany string v `admin.py`?

Reply:
@viliambalaz V admin rozhrani staci vsetko anglicky. Admin rozhranie netreba lokalizovat. Ak si vsimnes, tak v `django.po` ani nic z admin rozrania nie je.

Reply:
:ok: